ruby/lib/google/protobuf.rb

```diff
@@ -31,6 +31,15 @@
 # require mixins before we hook them into the java & c code
 require 'google/protobuf/message_exts'
 
+# We define these before requiring the platform-specific modules.
+# That way the module init can grab references to these.
+module Google
+  module Protobuf
+    class Error < RuntimeError; end
```

Comment:
Done. It's funny, I always wondered what the distinction between these was supposed to be. Way back in the day I couldn't find any docs to explain it so I asked on ruby-talk and no one answered:
https://groups.google.com/forum/#!topic/comp.lang.ruby/BGIpjSnC3Nc

Happy to finally have the difference explained. :)